docs: update application development content (#45)
equinor/dm-docs: docs/docs/guides/application-development.md

To run dependent services locally you can use the included `docker-compose.yaml` file. This file includes databases, DMSS service, and job service. 

Start services in the background (-d detached mode) by running:

```bash
docker-compose pull

> docker compose up --build -d

Reviewer: ```suggestion
docker compose up
```
There is nothing to build in this compose file. Every image is pulled.
Also, perhaps dont recommend running detached? Most people running locally will be interested in viewing the logs.

```

To start the app development server, run:

```bash
npm start
```

The web app can now be reached at [http://localhost:3000](http://localhost:3000) in the web browser. The web page will automatically reload if you make changes to the code.
You will see the build errors and lint warnings in the console.

:::note 

Remember: You must have the docker-compose services running to use the web application.

:::

## Reset data

We recommend create a virtual Python environment. 

Create a new virtual environment by running: 

```bash
python3 -m venv .venv
``` 

> Activate the virtual environment by running:

Reviewer: Perhaps remove all this python/venv stuff?
Will propably not apply to many environments (like windows in Equinor) anyway.
Just shorten in to "Install the python package dm-cli"

Author: I made it a tip (in a accordion), so that it is not that visible 👍

<details>
<summary>Linux</summary>

```shell
$ source .venv/bin/activate
```
</details>

<details>
<summary>Windows</summary>

```shell
$ .\venv\Scripts\Activate.ps1
$ pip install --upgrade pip
```

</details>

Install the dm-cli package (available on [PyPi](https://pypi.org/project/dm-cli/)) by running: 

```bash
pip install -r requirements.txt 
```

Run this command to reset the application data (upload any entities, blueprints, and recipes registered under the `app/` folder) to the data modelling storage service (DMSS):

```bash
./reset-app.sh
```

:::note

This script must be run every time a modification is done in any files in the `app/`  folder.

:::

## Build production version

When you’re ready to deploy to production, create a minified bundle by running the command:

```bash
npm run build
```

## Configure the app

> This template uses an application entity to control the behavior of the app. This entity can be seen at `app/data/DemoApplicationDataSource/instances/demoApplication.json`. The `.env` points to this application entity, which will be loaded and used to select what UI plugin to be shown. The application recipe link that defines what to show is located at `app/data/DemoApplicationDataSource/DemoApplication/recipes/demoApp.json`. 

Reviewer: ```suggestion
This template uses an application entity to control the behavior of the app. This entity can be seen at `app/data/DemoApplicationDataSource/instances/demoApplication.json`. The `.env`-file has a variable pointing to this application entity, which will be loaded and used to select what _UIPlugin_ to be shown. The application _RecipeLink_ that defines what to show is located at `app/data/DemoApplicationDataSource/DemoApplication/recipes/demoApp.json`. 
```

## Extending the app

Add functionality and customize your apps using plugins. Apps is designed to be extensible, and one of the best ways to add functionality to apps is through the plugin system.

Add functionality and customize your apps using [plugins](./../concepts/plugins.md) and [recipes](./../concepts/recipes.md).

See [plugin development](./plugin-development.md) for creating and adding new plugins.

## Modelling domain

Add models using [blueprints](./../concepts/blueprints.md).

See [domain modeling](./domain-modeling.md) for how to create and add models for the app.

## Pre commit 

To ensure consistent code formatting and to do a minimum of checks on the local computer before committing code to a Git repository, the app provides a `.pre-commit-config.yaml` file that is used to setup Git _pre-commit hooks_.
The pre-commit hook will run [prettier](https://prettier.io/) formatting, certain tests, sanitisers and analyze the JavaScript code using [eslint](https://eslint.org/).
To run pre-commit, it needs to be installed on your local machine with

```bash
pip install pre-commit
```

> This tell pre-commit to always run for this repository on every commit.

Reviewer: "pre-commit install" is missing here, if you want it to run automatically
